langchain4j-agentic/src/test/java/dev/langchain4j/agentic/SupervisorAgentIT.java

```diff
@@ -511,4 +521,127 @@ void typed_banker_with_memory_test() {
+                .build();
+
+        GeneralAssistant generalAssistant = AgenticServices.agentBuilder(GeneralAssistant.class)
+                .chatModel(baseModel())
+                .chatMemoryProvider(memoryId -> new InMemoryChatMemoryProvider("general", messageMap))
+                .build();
+
+        Supervisor supervisorAgent = AgenticServices.supervisorBuilder(Supervisor.class)
+                .subAgents(generalAssistant, jokesterAssistant)
+                .chatModel(plannerModel())
+                .chatMemoryProvider(memoryId -> new InMemoryChatMemoryProvider("supervisor", messageMap))
+                .responseStrategy(SupervisorResponseStrategy.LAST)
+                .build();
+
+        String lionJoke1 = supervisorAgent.respond("supervisor", "Tell me a joke about lions");
+        assertThat(lionJoke1).isNotNull().containsIgnoringCase("lion");
+
+        // Simulate recreating the same functional Supervisor System, same memory and all, new unique names will be generated
+        JokesterAssistant jokesterAssistant2 = AgenticServices.agentBuilder(JokesterAssistant.class)
+                .chatModel(baseModel())
+                .chatMemoryProvider(memoryId -> new InMemoryChatMemoryProvider("jokester", messageMap))
+                .build();
+
+        GeneralAssistant generalAssistant2 = AgenticServices.agentBuilder(GeneralAssistant.class)
+                .chatModel(baseModel())
+                .chatMemoryProvider(memoryId -> new InMemoryChatMemoryProvider("general", messageMap))
+                .build();
+
+        Supervisor supervisorAgent2 = AgenticServices.supervisorBuilder(Supervisor.class)
+                .subAgents(generalAssistant2, jokesterAssistant2)
+                .chatModel(plannerModel())
+                .chatMemoryProvider(memoryId -> new InMemoryChatMemoryProvider("supervisor", messageMap))
+                .responseStrategy(SupervisorResponseStrategy.LAST)
+                .build();
+
+        String lionJoke2 = supervisorAgent2.respond("supervisor", "tell me a joke about cheetahs");
+        assertThat(lionJoke2).isNotNull().containsIgnoringCase("cheetah");
+
+        List<ChatMessage> supervisorMessages = supervisorAgent.getChatMemory("supervisor").messages();
+
+        Set<String> agentNames = supervisorMessages.stream()
+                .filter(AiMessage.class::isInstance)
+                .map(AiMessage.class::cast)
+                .map(AiMessage::text)
+                .map(text -> {
+                    int start = text.indexOf("\"agentName\":") + "agentName:".length();
+                    int agentNameStart = text.indexOf('"', start + 1)+1;
+                    return text.substring(agentNameStart, text.indexOf('"', agentNameStart + 1));
+                }).collect(Collectors.toSet());
+
+        // only 2 agents, the jokester and done
+        assertThat(agentNames).hasSize(2);
+        assertThat(agentNames).contains("done");
```

Comment:
```suggestion
        assertThat(agentNames).containsExactly("jokester", "done");
```
nit: the comment specifies which ones are expected, perhaps worth checking exactly?